addons/l10n_ec/models/l10n_latam_document_type.py

```diff
@@ -0,0 +1,41 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class L10nLatamDocumentType(models.Model):
+    _inherit = 'l10n_latam.document.type'
+    
+    _EC_TYPE = [
+        ('out_invoice', 'Customer Document'),
+        ('in_invoice', 'Supplier Document'),
+        ('out_refund', 'Customer Refund'),
+        ('in_refund', 'Supplier Refund'),
+        ('out_waybill', 'Issued Waybill'),
+        ('out_withhold', 'Customer Withhold'),
+        ('in_withhold', 'Supplier Withhold'),
+        ('hr_advance', 'Employee Advance'),
+        ('other', 'Others'),
+    ]
+
+    _EC_AUTHORIZATION = [
+        ('none', 'None'),
+        ('own', 'Issued by my company'),
+        ('third', 'Issued by Third Parties')
+    ]
+
+    l10n_ec_type = fields.Selection(
+        _EC_TYPE, 
+        string='Ecuadorian Type',
+        help='Indicates the aplicability of the document',
+        )
+    l10n_ec_require_vat = fields.Boolean(
```

Comment:
@Stentor check if we can use the xml_id of the document type so we don't use this new field.

Reply:
Same as above

We need a mechanism to diferentiate document types according to move type,so the user doesnt use a wrong document type according to the account_move type.

 Exemple:

&lt;record id="ec_04" model="l10n_latam.document.type"&gt;
			&lt;field name="code"&gt;18&lt;/field&gt;
			&lt;field name="name"&gt;Factura de Venta&lt;/field&gt;
			&lt;field name="report_name"&gt;Factura&lt;/field&gt;
			&lt;field name="doc_code_prefix"&gt;Fact&lt;/field&gt;
			&lt;field name="l10n_ec_require_vat"&gt;True&lt;/field&gt;
			&lt;field name="sequence"&gt;15&lt;/field&gt;
			&lt;field name="internal_type"&gt;invoice&lt;/field&gt;
			&lt;field name="l10n_ec_type"&gt;out_invoice&lt;/field&gt;
			&lt;field name="l10n_ec_authorization"&gt;own&lt;/field&gt;
			&lt;field name="country_id" ref="base.ec"/&gt;
		&lt;/record&gt;

&lt;record id="ec_06" model="l10n_latam.document.type"&gt;
			&lt;field name="code"&gt;01&lt;/field&gt;
			&lt;field name="name"&gt;Factura de Compra&lt;/field&gt;
			&lt;field name="report_name"&gt;Factura&lt;/field&gt;
			&lt;field name="doc_code_prefix"&gt;Fact&lt;/field&gt;
			&lt;field name="l10n_ec_require_vat"&gt;True&lt;/field&gt;
			&lt;field name="sequence"&gt;10&lt;/field&gt;
			&lt;field name="internal_type"&gt;invoice&lt;/field&gt;
			&lt;field name="l10n_ec_type"&gt;in_invoice&lt;/field&gt;
			&lt;field name="l10n_ec_authorization"&gt;third&lt;/field&gt;
			&lt;field name="country_id" ref="base.ec"/&gt;
		&lt;/record&gt;

The field 10n_ec_require_vat can be removed, though.